libecole/include/ecole/environment/environment.hpp

```diff
@@ -109,10 +116,16 @@ class Environment {
 			reward_function().before_reset(model());
 			information_function().before_reset(model());
 			auto const [done, action_set] = dynamics().reset_dynamics(model(), std::forward<Args>(args)...);
-
 			can_transition = !done;
+
+			OptionalObservation observation;
+			if (!done) {
+				observation = observation_function().extract(model(), done);
+			} else {
+				observation = OptionalObservation{};
+			}
```

Comment:
I think the `else` here is redundant, as `observation` is already instantiated to an empty observation when it is declared.. What about
```suggestion
			auto observation = OptionalObservation{};
			if (!done) {
				observation = observation_function().extract(model(), done);
			}
```